### PR TITLE
WIP: added configuration for incoming email

### PR DESCRIPTION
**Pull request by denis-ev**
Here is the PR @timabbott,

haven't changed much, just tried to follow the documentation [Documentation Link](https://zulip.readthedocs.io/en/latest/production/email-gateway.html).

[Zulip Ref](https://chat.zulip.org/#narrow/stream/3-backend/topic/docker.20incoming.20email)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ WORKDIR /home/zulip
 # You can specify these in docker-compose.yml or with
 #   docker build --build-args "ZULIP_GIT_REF=git_branch_name" .
 ARG ZULIP_GIT_URL=https://github.com/zulip/zulip.git
-ARG ZULIP_GIT_REF=2.0.5
+ARG ZULIP_GIT_REF=master
```
Comment (timabbott) [inline Dockerfile]: Leave off this hunk; it's unrelated to the other changes here.

**Comment by timabbott** [comment]
@denis-ev how have you tested this?  I worry that with this implementation, postfix won't come back up after the first reboot.  I suspect the better solution would be to patch `dockervoyager.pp` similar to how we manage `nginx`.

**Comment by denis-ev** [comment]
> @denis-ev how have you tested this? I worry that with this implementation, postfix won't come back up after the first reboot. I suspect the better solution would be to patch `dockervoyager.pp` similar to how we manage `nginx`.

@timabbott 
You’re right, postfix won’t start and when manually starting postfix it rejects messages. 
Mailbox not found, I’m working on it and let you now when it runs smoothly. 
I’m just not too much into postfix, any idea why the messages are getting rejected?

**Comment by timabbott** [comment]
If I had to guess, the `mailname` for the system is wrong post reboot, or something like that.  The logs in `/var/log/mail.log` should explain exactly why.

**Comment by denis-ev** [comment]
@timabbott 
I've checked mail name before and I don't seem to find incoming emails at all.
When checking with telnet my server is connecting to postfix within the Zulip container.
I can't find a "mail.log", tho.

When the basic mail function is working, I'll look into keeping postfix running after a restart.

**Pull request closed without merging**